src/assignment-4/YearnVaultMock.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {ERC20Permit} from "yield-utils-v2/token/ERC20Permit.sol";
+import {IERC20} from "yield-utils-v2/token/IERC20.sol";
+
+interface IYearnVault {
+    function deposit(uint256 amount) external returns (uint256);
+    function withdraw(uint256 maxShares) external returns (uint256);
+    // function pricePerShare() external view returns (uint256);
+}
+
+contract YearnVaultMock is IYearnVault, ERC20Permit {
+    
+    uint256 constant RAY = 10 ** 27;
+    uint256 public pricePerShareMock = RAY;
+    IERC20 public immutable token;
+
+    error TransferFailed();
+
+    constructor(address token_, string memory name, string memory symbol, uint8 decimals) ERC20Permit(name, symbol, decimals) {
+        token = IERC20(token_);
+    }
+
+    function pricePerShare() external view returns (uint256) {
```

Comment:
So, the mocks don't get natspec love?

Reply:
:+1: fixed: https://github.com/davidbrai/yield-mentorship/pull/5/commits/93c82b91814228389e267b81dcc921fd9c5e5ffe